server/model.go

```diff
@@ -361,23 +383,23 @@ func (m *Model) parseToolCalls(s string) ([]api.ToolCall, bool) {
 			return nil, false
 		} else {
 			offset += int(decoder.InputOffset())
-			objs = append(objs, obj)
+			objs = append(objs, collect(obj)...)
 		}
 	}
 
 	var toolCalls []api.ToolCall
 	for _, kv := range objs {
-		var call api.ToolCall
-		for k, v := range kv {
-			switch k {
-			case name:
-				call.Function.Name = v.(string)
-			case arguments:
-				call.Function.Arguments = v.(map[string]any)
+		n, nameOk := kv[name].(string)
+		a, argumentsOk := kv[arguments].(map[string]any)
+		if nameOk && argumentsOk {
```

Comment:
AFAIK `arguments` is not optional, so this checks for both in the returned tool call to avoid false positives